lib/Widgets/Settings.vala

```diff
@@ -96,6 +115,40 @@ namespace Granite {
             }
         }
 
+        private void setup_accent_color () {
+            try {
+                portal = Portal.Settings.get ();
+
+                var variant = portal.read (
+                    "org.freedesktop.appearance",
+                    "accent-color"
+                ).get_variant ();
+
+                update_color (variant);
+
+                portal.setting_changed.connect ((scheme, key, value) => {
+                    if (scheme == "org.freedesktop.appearance" && key == "accent-color") {
+                        update_color (value);
+                    }
+                });
+
+                return;
+            } catch (Error e) {
+                debug (e.message);
+            }
+
+            // Set a default in case we can't get from system
+            accent_color = "#3689e6";
+        }
+
+        private void update_color (GLib.Variant color) {
+            double red, green, blue;
+            color.get ("(ddd)", out red, out green, out blue);
+
+            Gdk.RGBA rgba = {(float) red, (float) green, (float) blue, 1};
+            accent_color = rgba.to_string ();
+        }
```

Comment:
I think we should keep this method just parse Variant representation to string representation of color and substitute to `accent_color` in the caller side instead.

```vala
        private string parse_color (GLib.Variant color) {
            double red, green, blue;
            color.get ("(ddd)", out red, out green, out blue);

            Gdk.RGBA rgba = {(float) red, (float) green, (float) blue, 1};
            return rgba.to_string ();
        }
```